### PR TITLE
Move FxCop\Desktop.Analyzers project over

### PR DESCRIPTION
**Pull request by genlu**
- Copy Desktop.Analyzers from [Roslyn repo](https://github.com/dotnet/roslyn).
- Update all projects to use shared targets/settings files, and references to correct NuGet packages.
- Add projects to the solution.
- Fix `NuGetRestore.ps1` to restore for entire solution.

@tmeschter @srivatsn

### CODE REVIEW

File: build/Targets/Analyzers.Versions.targets [hunk 1]
```diff
@@ -9,6 +9,9 @@
     <MicrosoftCodeAnalysisAnalyzersSemanticVersion>1.0.1</MicrosoftCodeAnalysisAnalyzersSemanticVersion>
     <MicrosoftCodeAnalysisAnalyzersPreReleaseVersion>beta1</MicrosoftCodeAnalysisAnalyzersPreReleaseVersion>
 
+    <DesktopAnalyzersSemanticVersion>1.0.1</DesktopAnalyzersSemanticVersion>
```
Comment (tmeschter) [inline build/Targets/Analyzers.Versions.targets]: As we have yet to release Desktop.Analyzers, the first version should be 1.0.0.

File: src/FxCop/Desktop.Analyzers/CSharp/CSharpDesktopAnalyzers.csproj [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\Core\DesktopAnalyzers.csproj">
+      <Project>{D1B42764-D442-4E4C-B2E7-9BA36FFBDDF1}</Project>
+      <Name>DesktopAnalyzers</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <Analyzer Include="..\..\..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll" />
+    <Analyzer Include="..\..\..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.0.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll" />
```
Comment (tmeschter) [inline src/FxCop/Desktop.Analyzers/CSharp/CSharpDesktopAnalyzers.csproj]: These analyzers appear to be duplicates.

**Comment by tmeschter** [comment]
I see there are three .props files, which all seem to be identical. Shouldn't there be just one?

**Comment by genlu** [comment]
@tmeschter Yes, they are. But they can become different at some point (e.g. language agnostic analyzers, only have analyzer for one language, etc.), maybe I should keep them as is right now?

**Comment by tmeschter** [comment]
@genlu In general, we won't accept pull requests for changes/new analyzers if they only support one language. Regardless, for the time being these files are all the same, so we should only have one.

**Comment by genlu** [comment]
@tmeschter Alright, fixed.

**Comment by tmeschter** [comment]
:+1:

**Pull request merged**